Mage/src/main/java/mage/abilities/effects/ReturnMORToBattlefieldUnderOwnerControlWithCounterEffect.java

```diff
@@ -0,0 +1,128 @@
+package mage.abilities.effects;
+
+import mage.MageObjectReference;
+import mage.abilities.Ability;
+import mage.cards.Card;
+import mage.cards.MeldCard;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.constants.Zone;
+import mage.counters.Counter;
+import mage.game.Game;
+import mage.game.events.EntersTheBattlefieldEvent;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class ReturnMORToBattlefieldUnderOwnerControlWithCounterEffect extends OneShotEffect {
+
+    private final MageObjectReference objectToReturn;
+    private final Set<Counter> counters;
+    private final String counterText;
+
+    public ReturnMORToBattlefieldUnderOwnerControlWithCounterEffect(
```

Comment:
Rather than requiring other cards to make a static `Set<Counter>` for a single counter instance, I'd prefer to offer an additional constructor that takes a `Counter` and does the set creation.

Also, is there a reason to use `Set<Counter>` rather than the `Counters` class?

Reply:
I think I tried to do that at some point but was confused by what the keys were in the `Counters`'s map.
I'll give it another look.

And indeed for the single Counter constructor, that would be better.